File: src/java/ReverseString.java
import java.util.Stack;

/**
 * @author LJY
 */
public class ReverseString {

    public String reverseString(String s) {
        if(s == null || "".equals(s)){
            return s;
        }
        char[] chars = s.toCharArray();
        Stack<Character> stack = new Stack<>();
        for (int i = chars.length - 1; i >= 0; i--) {
            stack.push(chars[i]);
        }
        StringBuilder stringBuilder = new StringBuilder(chars.length);
        stack.forEach(e -> stringBuilder.append(e.charValue()));
        return stringBuilder.toString();
    }

    public static void main(String[] args) {
        String s = new ReverseString().reverseString("A man, a plan, a canal: Panama");
        System.out.println(s.equals("amanaP :lanac a ,nalp a ,nam A"));
    }

}

File: src/scala/ReverseString.scala
import java.util

object ReverseString {

  def reverseString(s: String): String = {
    if (s == null || "" == s) return s
    var chars = s.toCharArray
    var stack = new util.Stack[Character]
    for (i <- chars.length - 1 to 0) {
      stack.push(chars(i))
    }
    val stringBuilder = new StringBuilder(chars.length)
    stack.forEach((e: Character) => stringBuilder.append(e.charValue))
    stringBuilder.toString
  }

}
